internal/pkg/service/common/etcdop/iterator/iterator_typed.go

```diff
@@ -38,7 +39,37 @@ func (v DefinitionT[T]) Do(ctx context.Context, opts ...op.Option) *IteratorT[T]
 
 // ForEachOp method converts iterator to for each operation definition, so it can be part of a transaction.
 func (v DefinitionT[T]) ForEachOp(fn func(value T, header *Header) error) *ForEachOpT[T] {
-	return &ForEachOpT[T]{def: v, fn: fn}
+	return &ForEachOpT[T]{def: v, onValue: fn}
+}
+
+// WithResultTo method converts iterator to for each operation definition, so it can be part of a transaction.
+func (v DefinitionT[T]) WithResultTo(slice *[]T) *ForEachOpT[T] {
+	return v.
+		ForEachOp(func(value T, header *Header) error {
+			*slice = append(*slice, value)
+			return nil
+		}).
+		AndOnFirstPage(func(response *etcd.GetResponse) error {
+			*slice = nil
+			return nil
+		})
+}
```

Comment:
I need to use iterator together with `AtomicOp`, so I need to store all results to the slice defined by the pointer.
Therefore I added `WithResultTo` method.

Reply:
`AtomicOp` performs retries on transaction failures, so at the beginning I have to reset the slice.
Therefore, I create `AndOnFirstPage` method.